### PR TITLE
Use the theme's webpack configuration if it exists.

### PR DESCRIPTION
**Pull request by devbucket**

### CODE REVIEW

File: lib/app/frontend/FrontendProcess.js [hunk 1]
```diff
@@ -55,12 +56,47 @@ class FrontendProcess {
...
    * Runs the Webpack Dev Server.
    */
   async webpackDevServer () {
+    const appFolder = this.appSettings.getApplicationFolder();
+    const themePath = resolve(appFolder, 'themes', this.options.theme)
+
+    /**
+     * Use theme own webpack server implementation if server config file is present
+     */
+    if (existsSync(resolve(themePath, 'webpack.config.js'))) {
```
Comment (fkloes) [inline lib/app/frontend/FrontendProcess.js]: Right now it's not possible to use your 2nd pull request together with the latest `platform-sdk 1.10.0`. That causes a direct dependency between the sdk and Engage. After merging this PR, we'd need to release an @alpha version of the sdk to enable development of pwa 6.11.0

Bildschirmfoto 2019-11-27 um 08 51 23

File: lib/app/frontend/FrontendProcess.js [hunk 2]
```diff
@@ -55,12 +56,47 @@ class FrontendProcess {
...
   }
 
+  /**
+   * Sleep N ms
+   * @param ms
+   * @returns {*}
+   */
+  async sleep (ms) {
+    return new Promise((resolve) => {
+      setTimeout(resolve, ms);
```
[the remark this answers is not in the document]
Comment (devbucket) [reply lib/app/frontend/FrontendProcess.js]: Done

File: lib/app/frontend/FrontendProcess.js [hunk 3]
```diff
@@ -55,12 +56,47 @@ class FrontendProcess {
...
    */
   async webpackDevServer () {
+    const appFolder = this.appSettings.getApplicationFolder();
+    const themePath = resolve(appFolder, 'themes', this.options.theme)
+
+    /**
+     * Use theme own webpack server implementation if server config file is present
+     */
+    if (existsSync(resolve(themePath, 'webpack.config.js'))) {
+      await this.sleep(400);
```
[the remark this answers is not in the document]
Comment (devbucket) [reply lib/app/frontend/FrontendProcess.js]: Done

File: lib/app/frontend/FrontendProcess.js [hunk 4]
```diff
@@ -55,12 +56,47 @@ class FrontendProcess {
...
+    return new Promise((resolve) => {
+      setTimeout(resolve, ms);
+    })
+  }
+
   /**
    * Runs the Webpack Dev Server.
    */
   async webpackDevServer () {
+    const appFolder = this.appSettings.getApplicationFolder();
```
[the remark this answers is not in the document]
Comment (devbucket) [reply lib/app/frontend/FrontendProcess.js]: Done

**Pull request merged**